File: C/gemma2-9b-it.c
#include <stdio.h>

// Function to swap two elements
void swap(int *a, int *b) {
    int temp = *a;
    *a = *b;
    *b = temp;
}

// Function to perform Bubblesort
void bubblesort(int array[], int n) {
    for (int i = 0; i < n - 1; i++) {
        // Flag to track if any swaps were made in the current iteration
        int swapped = 0;
        
        // Iterate through the unsorted portion of the array
        for (int j = 0; j < n - i - 1; j++) {
            // If the current element is greater than the next element, swap them
            if (array[j] > array[j + 1]) {
                swap(&array[j], &array[j + 1]);
                swapped = 1;
            }
        }
        
        // If no swaps were made in the current iteration, the array is already sorted
        if (!swapped) {
            break;
        }
    }
}

// Example usage with 1000 inputs
int main() {
    int array[1000];
    for (int i = 0; i < 1000; i++) { //adicionar no prompt que os valores precisam ser aleatórios
        array[i] = i;
    }
    
    // Reverse the array to make it unsorted
    for (int i = 0; i < 500; i++) {
        swap(&array[i], &array[999 - i]);
    }
    
    bubblesort(array, 1000);
    // No print statements as per the requirement
    return 0;
}

File: Python/gemma2-9b-it.py
def bubblesort(arr):
    """
    Optimized implementation of the Bubblesort algorithm.

    Args:
        arr (list): The list of elements to be sorted.

    Returns:
        list: The sorted list of elements.
    """
    n = len(arr)

    for i in range(n):
        # Create a flag that will allow the function to terminate early if there's nothing left to sort
        swapped = False

        # Start looking at each item of the list one by one, comparing it with its adjacent value
        for j in range(0, n - i - 1):
            # If we find an element that is greater than its adjacent element, then swap them
            if arr[j] > arr[j + 1]:
                # Swap values
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                # Set the flag to True so we'll loop again after this iteration
                swapped = True

        # If no two elements were swapped in the inner loop, the list is sorted
        if not swapped:
            break

    return arr


# Example usage with 1000 inputs
import random
arr = [random.randint(0, 1000) for _ in range(1000)]
sorted_arr = bubblesort(arr)